command/agent/agent.go

```diff
@@ -445,3 +470,67 @@ func (a *Agent) Stats() map[string]map[string]string {
 	}
 	return stats
 }
+
+func (a *Agent) createConsulConfig() {
+	cfg := &consul.ConsulConfig{
+		Addr:      a.config.ConsulConfig.Addr,
+		Token:     a.config.ConsulConfig.Token,
+		Auth:      a.config.ConsulConfig.Auth,
+		EnableSSL: a.config.ConsulConfig.EnableSSL,
+		VerifySSL: a.config.ConsulConfig.VerifySSL,
+		CAFile:    a.config.ConsulConfig.CAFile,
+		CertFile:  a.config.ConsulConfig.CertFile,
+		KeyFile:   a.config.ConsulConfig.KeyFile,
+	}
+	a.consulConfig = cfg
+}
+
+// syncAgentServicesWithConsul syncs the client and server services with Consul
+func (a *Agent) syncAgentServicesWithConsul(clientHttpAddr string, serverHttpAddr string) error {
+	cs, err := consul.NewConsulService(a.consulConfig, a.logger)
+	if err != nil {
+		return err
+	}
+	a.consulService = cs
+	var services []*structs.Service
+	addrs := make(map[string]string)
+	if a.client != nil && a.config.ConsulConfig.ClientServiceName != "" {
+		if err != nil {
+			return err
+		}
+		clientService := &structs.Service{
+			Name:      a.config.ConsulConfig.ClientServiceName,
+			PortLabel: "clienthttpaddr",
```

Comment:
why does this have to be a label that we look up in the map? just put the actual value in the string